src/networking/NetworkService.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * The NetworkService class defines the `getInstance` method that lets clients access
+ * the unique NetworkService instance. I wanted to do this since we only want/need 1 firebase instance.
+ */
+import { initializeApp } from "firebase/app";
+import {
+  FirebaseFirestore,
+  getFirestore,
+  getDoc,
+  doc,
+  initializeFirestore,
+  Settings,
+} from "firebase/firestore";
+import { NetworkResponse } from "./NetworkResponse";
+import {
+  PopulationOptions,
+  Populate,
+  COLLECTION_NAME,
+  getCollectionForReference,
+} from "./PopulationOptions";
+
+export class NetworkService {
+  private static instance: NetworkService;
+  private static db: FirebaseFirestore;
+  /**
+   * The NetworkService's constructor should always be private to prevent direct
+   * construction calls with the `new` operator.
+   */
+  private constructor() {
+    // initialize firebase
+    const firebaseApp = initializeApp({
+      projectId: "cdp-jackson-dev-002",
```

Comment:
Can this be parametrizable somehow? How should we integrate this with the cookiecutter? A JSON config somewhere or just a value somewhere?

Reply:
Just from what you have said, it should probably write a JSON file with this (and other) values for us to pick up. That sounds easy enough to utilize on our end. Would that be a problem?

Reply:
Nope totally fine!